personal/privacyidea/class_PrivacyIdeaUtils.inc

```diff
@@ -176,6 +178,10 @@ class PrivacyIdeaUtils implements PILog
         assert('string' === gettype($description));
         assert($tokenType == "paper" || $tokenType == "totp" || $tokenType == "webauthn");
 
+        if (!$this->authenticate()) {
+            return array();
+        }
+
```

Comment:
dito here

Reply:
This is already an existing  failure mode. However while reviewing this again I found a bug in `personal/privacyidea/tokens/class_MFAPaperToken.inc` where an error wasn't handled properly. I've fixed that in another commit.